stacks/api/function/enqueuer/src/http.ts

```diff
@@ -0,0 +1,150 @@
+import {Middlewares} from "@spica-server/core";
+import {EventQueue, HttpQueue} from "@spica-server/function/queue";
+import {Event, Http} from "@spica-server/function/queue/proto";
+import {Description, Enqueuer} from "./enqueuer";
+import express = require("express");
+import read = require("body-parser/lib/read");
+import bodyParser = require("body-parser");
+
+export class HttpEnqueuer extends Enqueuer<HttpOptions> {
+  description: Description = {
+    icon: "http",
+    name: "http",
+    title: "Http",
+    description: "Designed for APIs and Http Streaming"
+  };
+
+  private router = express.Router({mergeParams: true});
+
+  constructor(private queue: EventQueue, private http: HttpQueue, httpServer: express.Application) {
+    super();
+    this.router.use(
+      bodyParser.raw({
+        limit: "10mb",
+        type: "*/*"
+      })
+    );
+    this.router.use(this.handleUnhandled);
+    const stack = httpServer._router.stack;
+    httpServer.use("/fn-execute", this.router);
+    const expressInitIndex = stack.findIndex(l => l.name === "expressInit");
+    const layer = stack.splice(stack.length - 1, 1)[0];
+    stack.splice(expressInitIndex + 1, 0, layer);
+  }
+
+  private handleUnhandled(req, res) {
+    // By default express sends a default response if the OPTIONS route is unhandled
+    // https://github.com/expressjs/express/blob/3ed5090ca91f6a387e66370d57ead94d886275e1/lib/router/index.js#L640
+    if (!req.route) {
+      res.status(404).send({
+        message: "Invalid route",
+        url: req.originalUrl,
+        method: req.method,
+        engine: "Function"
+      });
+    }
+  }
+
+  private reorderUnhandledHandle() {
+    this.router.stack.splice(this.router.stack.findIndex(l => l.handle == this.handleUnhandled), 1);
+    this.router.use(this.handleUnhandled);
+  }
+
+  subscribe(target: Event.Target, options: HttpOptions): void {
+    const method = options.method.toLowerCase();
+    const path = options.path.replace(/^\/?(.*?)\/?$/, "/$1");
+
+    if (
+      options.preflight &&
+      options.method != HttpMethod.Get &&
+      options.method != HttpMethod.Head
+    ) {
+      const fn = (req, res, next) => Middlewares.Preflight(req, res, next);
+
+      Object.defineProperty(fn, "target", {writable: false, value: target});
+
+      this.router.options(path, fn);
+    }
+
+    if (options.method == HttpMethod.Options) {
+      throw new Error("Preflight have been used with with HttpMethod.Options");
+    }
```

Comment:
@thesayyn I guess this should throw error when request is preflight and request method is 'OPTION' but this if block is missing preflight case.